streams/src/test/java/org/apache/kafka/streams/state/internals/AbstractDualSchemaRocksDBSegmentedBytesStoreTest.java

```diff
@@ -251,7 +250,7 @@ public void shouldPutAndFetch() {
             // all records expired as actual from is 59001 and to is 1000
             final List<KeyValue<Windowed<String>, Long>> expected = Collections.emptyList();
 
-            assertEquals(expected, toList(values));
+            assertEquals(expected, toListAndCloseIterator(values));
```

Comment:
line#246 already use try-resource to handle the close, so why does `toListAndCloseIterator` need to call close again?

Reply:
You are right that it does not need to close the iterator, but it would be unnecessary code duplication to have both `toList` and `toListAndCloseIterator`...

And it does not "hurt" to close it twice.

It was just easier to refactor `toList` across the broad, instead of add `try-with-resources` to all callers which don't close the passed in iterator.

Thoughts? Also happy to do it differently.

Reply:
Maybe we could remove the try-with-resources to match the style used in AbstractRocksDBWindowStoreTest. For example:
```
        // For all tests, actualFrom is computed using observedStreamTime - retention + 1.
        // so actualFrom = 60000(observedStreamTime) - 1000(retention) + 1 = 59001
        // all records expired as actual from is 59001 and to is 1000
        assertEquals(Collections.emptyList(), toListAndCloseIterator(bytesStore.fetch(
                Bytes.wrap(keyA.getBytes()), 0, windows[2].start())));

        // all records expired as actual from is 59001 and to is 1000
        assertEquals(Collections.emptyList(), toListAndCloseIterator(bytesStore.fetch(
                Bytes.wrap(keyA.getBytes()), Bytes.wrap(keyB.getBytes()), 0, windows[2].start())));
```

Reply:
Lots of reformatting, but ok.